core/domain/skill_services.py

```diff
@@ -576,7 +576,7 @@ def save_skill_rights(skill_rights, committer_id, commit_message, commit_cmds):
     model.commit(committer_id, commit_message, commit_cmd_dicts)
 
 
-def delete_skill(committer_id, skill_id, force_deletion=False):
+def delete_skill(committer_id, skill_id, force_deletion=True):
```

Comment:
Was there any reason for this change? Not against it, just wanted to know.

Reply:
It breaks if force_deletion is not set to True. I'm not really sure what the correct solution is.

It breaks because, after some debugging, it looks like NDB's get_all() query returns deleted elements as well. This means that when querying for multiple skills, it will look for skills that are already deleted and throw an error.

Specifically, this happens on the skills list in the topic and skills dashboard. There is a query on line 77 of topics_and_skills_dashboard.py that would return deleted skill IDs if force_deletion is set to false. This results in querying for deleted skills later on, breaking on line 93 when querying for skill summaries.

Reply:
NDB's get_all() would return all (including, force_deletion=false deletions), but we have defined a get_all() function in base_model/gae_models.py that takes in parameter include_deleted, with which we can filter out deleted skills.
Also, this include_deleted is set to False by default, and I don't think that is set to True in any calls that we are making.
So, I am not sure why this is happening.

Reply:
Okay, I changed the query to not included deleted items, and turned force_deletion off.